Sources/TokamakCore/Animation/_AnyAnimatableData.swift

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// A type-eraser for `VectorArithmetic`.
+public struct _AnyAnimatableData: VectorArithmetic {
+  private var box: _AnyAnimatableDataBox?
```

Comment:
SwiftUI implements this using an `_AnyAnimatableDataVTable` class, but I didn't find a reason for having this implementation complexity. From some quick testing, I found that `AnyShape(Circle()).animatableData` just have a `VTable<Circle>` implementation, while `_AnyAnimatableDataVTable.zero` uses `ZeroVTable`. This implementation, however, should be the same as it is with SwiftUI.